pkg/record/record.go

```diff
@@ -30,11 +32,29 @@ type Marshalable interface {
 	GetExtension() string
 }
 
+// managedFieldsSetter is a universal interface of
+// resources that implement the SetManagedFields method.
+type managedFieldsSetter interface {
+	SetManagedFields(managedFields []metav1.ManagedFieldsEntry)
+}
+
 type JSONMarshaller struct {
 	Object interface{}
 }
 
 func (m JSONMarshaller) Marshal(_ context.Context) ([]byte, error) {
+	if mfs, ok := m.Object.(managedFieldsSetter); ok {
+		mfs.SetManagedFields([]metav1.ManagedFieldsEntry{})
+	} else if strMap, ok := m.Object.(map[string]interface{}); ok {
```

Comment:
If we passed the `unstructured.Unstructured` struct instead of the `unstructured.Unstructured.Object` map, the first condition would hit and the map-related code wouldn't have to be there at all.

Reply:
I am failing to see how it would work with other types, which are not `Unstructured`. Also the `unstructured.Unstructured.Object` is not the same `Object` we have here...

Reply:
I'm sorry about the confusion. It's tricky to describe these kinds of ideas in text very well when everything is called `Object`.